Woof/Woof/ViewModel/LoginViewModel.swift

```diff
@@ -1,11 +1,76 @@
 import Foundation
 
-/// The object that changes the state of the data in the model based on user actions in `LoginView`.
-final class LoginViewModel {
-    /// Sets the current's user role in the app.
+/// An interface is responsible for preparing and providing data for the `LoginView`.
+final class LoginViewModel: ObservableObject {
+    // MARK: - Internal interface
+
+    /// Indicates whether the owner role is selected.
+    ///
+    /// Returns `true` if the owner role is selected, otherwise `false`.
+    ///
+    /// The property publishes updates when the value is changed.
+    @Published var isTheOwnerRoleSelected = false {
+        didSet {
+            if !isTheCurrentOwnerExists {
+                createAndSaveNewOwner()
+            }
+        }
+    }
+
+    /// Indicates whether the sitter role is selected.
+    ///
+    /// Returns `true` if the sitter role is selected, otherwise `false`.
     ///
-    /// - Parameter userRole: The role to set.
-    func set(_ userRole: Role) {
-        PreferencesHandler.set(userRole: userRole)
+    /// The property publishes updates when the value is changed.
+    @Published var isTheSitterRoleSelected = false {
+        didSet {
+            if !isTheCurrentSitterExists {
+                createAndSaveNewSitter()
+            }
+        }
+    }
+
+    // MARK: - Private interface
+
+    private var isTheCurrentOwnerExists: Bool {
+        guard let data = KeyValueStorage(KeyValueStorage.Name.currentOwner)
+            .loadData(for: KeyValueStorage.Key.currentOwner) else {
+            return false
+        }
+        guard (try? JSONDecoder().decode(Owner.self, from: data)) != nil else {
+            return false
+        }
+
+        return true
+    }
+
+    private var isTheCurrentSitterExists: Bool {
+        guard let data = KeyValueStorage(KeyValueStorage.Name.currentSitter)
+            .loadData(for: KeyValueStorage.Key.currentSitter) else {
+            return false
+        }
+        guard (try? JSONDecoder().decode(Sitter.self, from: data)) != nil else {
+            return false
+        }
+
+        return true
+    }
+
+    private func createAndSaveNewOwner() {
```

Comment:
nit:
using the word `and` in method naming is unusual. It is strange if we can save objects without creating.
```suggestion

    private func saveNewOwner() {
```

Reply:
Fixed.